react/src/components/EditableVFolderName.tsx

```diff
@@ -0,0 +1,182 @@
+  | ({ component: typeof Title } & Omit<TitleProps, 'children'>)
+);
+
+const EditableVFolderName: React.FC<EditableVFolderNameProps> = ({
+  component: Component = Text,
+  vfolderFrgmt,
+  editable: editableOfProps,
+  style,
+  enableLink = true,
+  onEditEnd,
+  ...otherProps
+}) => {
+  const vfolder = useFragment(
+    graphql`
+      fragment EditableVFolderNameFragment on VirtualFolderNode {
+        id
+        name
+        user
+        group
+        status
+      }
+    `,
+    vfolderFrgmt,
+  );
+  const [optimisticName, setOptimisticName] = useState(vfolder.name);
+  const [userInfo] = useCurrentUserInfo();
+  const currentProject = useCurrentProjectValue();
+  const baiClient = useSuspendedBackendaiClient();
+  const renameMutation = useTanMutation({
+    mutationFn: (input: { id: string; name: string }) => {
+      return baiClient.vfolder.rename(input.name, toLocalId(vfolder?.id));
+    },
+  });
+  const painKiller = usePainKiller();
+
+  const { t } = useTranslation();
+  const { token } = theme.useToken();
+  const { message } = App.useApp();
+  const { generateFolderPath } = useFolderExplorerOpener();
+  const [isEditing, setIsEditing] = useState(false);
+
+  const isEditingAllowed =
+    editableOfProps &&
+    (userInfo.uuid === vfolder.user || currentProject?.id === vfolder.group) &&
+    !isDeletedCategory(vfolder.status);
+
+  return (
+    <>
+      {(!isEditing || renameMutation.isPending) && (
+        <Component
+          editable={
+            isEditingAllowed && !renameMutation.isPending
+              ? {
+                  onStart: () => {
+                    setIsEditing(true);
+                  },
+                  triggerType: ['icon', 'text'],
+                }
+              : false
+          }
+          style={{
+            ...style,
+            color: renameMutation.isPending
+              ? token.colorTextTertiary
+              : style?.color,
+          }}
+          {...otherProps}
+        >
+          {enableLink && (
+            <BAILink
+              type="hover"
+              to={generateFolderPath(toLocalId(vfolder?.id))}
+            >
+              {renameMutation.isPending ? optimisticName : vfolder.name}
+            </BAILink>
+          )}
+          {!enableLink &&
+            (renameMutation.isPending ? optimisticName : vfolder.name)}
+        </Component>
+      )}
+      {isEditing && !renameMutation.isPending && (
+        <Form
+          onFinish={(values) => {
+            setIsEditing(false);
+            setOptimisticName(values.vfolderName);
+            renameMutation.mutate(
+              {
+                id: vfolder.id,
+                name: values.vfolderName,
+              },
+              {
+                onSuccess: () => {
+                  message.success(t('data.folders.FileRenamed'));
+                  onEditEnd?.(true);
+                },
+                onError: (error) => {
```

Comment:
Instead of using a callback prop to refetch, we can use Relay's `fetchQuery` to update some fields of `vfolder_node`. After `fetchQuery`, the vfolder list based on `vfolder_node` will be updated automatically by Relay. Please refer to https://github.com/lablup/backend.ai-webui/blob/88f1c2bed2cc0164891ded0068e7e33159419af5/react/src/components/ComputeSessionNodeItems/TerminateSessionModal.tsx#L274-L291